### ISSUE TITLE
Unexpected CS8602 warning

### ISSUE THREAD

**Issue by krt-** [dotnet/roslyn#48553]
**Version Used**: 
5.0.0-rc.1.20451.14

**Steps to Reproduce**:

``` csharp
using System.Collections.Generic;
using System.Linq;

#nullable enable

public class Foo { public int Bar { get; set; } }

public class C {
    public void M() {
        var list = new List<Foo> { new Foo() };        
        var foo = list.FirstOrDefault();
        var found = foo is {};

        if (!found)
        {
            foo = new Foo();
        }
        
        foo.Bar = 1;    //CS8602 warning    
    }
}
```

There's no warning if `found` variable is inlined, e.g. `if (!(foo is { }))`

**Expected Behavior**:
No CS8602 warning

**Actual Behavior**:
CS8602 warning

**Comment by Joe4evr** [comment]
[Copied repro into SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAN0QSwGYE8A+AAgEwCMAsAFCEAMABIaQCwDcVtDpAdADKYB2ARzaUqAYn4BXADbSIwaXDpx+8xVXYBmBsToAxAPYG6AbwbaBMOgCEICU3QDmcGCzpQXbgL50vG6tokdADCVCZUdJHmDEx0ALIAFACUEVHhlFGZdGh2dNKYsHQAvHT8cADudHywADyGBgB8DmWV9cm+blmRqV059thGxXkFMFx6mAiwAPIIACJw2BAyMMkiXZF9dAOS/AAmQwPGBaZeIj1ZOHQJAITbeykZ6wD0T5c3CYd0xyZeSQ/rpnOAM+JRa+iMqyBUT8jy6UMihy4tnsJVInUiL2CAGUABwANhounKdn4AkcmXOMK8QA=) for easy viewing.

**Comment by jaredpar** [comment]
Thanks for taking the time to file this issue. This behavior is "By Design". The flow tracking in the compiler does not track nullable state associated with `bool` values in this way.
